src/main/java/uk/gov/ida/matchingserviceadapter/MatchingServiceAdapterModule.java

```diff
@@ -187,10 +189,31 @@ public EidasMatchingDatasetUnmarshaller getEidasMatchingDatasetUnmarshaller() {
     @Singleton
     public EidasUnsignedMatchingDatasetUnmarshaller getEidasUnsignedMatchingDatasetUnmarshaller(
             SecretKeyDecryptorFactory secretKeyDecryptorFactory,
-            MetadataResolverRepository eidasMetadataResolverRepository) {
+            MetadataResolverRepository eidasMetadataResolverRepository,
+            InstantValidator instantValidator,
+            SubjectValidator subjectValidator,
+            CountryConditionsValidator conditionsValidator,
+            @Named("AllAcceptableHubConnectorEntityIds") List<String> acceptableHubConnectorEntityIds
+            ) {
         StringToOpenSamlObjectTransformer<Response> stringtoOpenSamlObjectTransformer = new CoreTransformersFactory().getStringtoOpenSamlObjectTransformer(new ResponseSizeValidator());
         EidasValidatorFactory eidasValidatorFactory = new EidasValidatorFactory(eidasMetadataResolverRepository);
-        return new EidasUnsignedMatchingDatasetUnmarshaller(secretKeyDecryptorFactory, stringtoOpenSamlObjectTransformer, eidasValidatorFactory);
+
+        ExceptionThrowingValidator<Assertion> eidasSamlValidator = a -> {
+            try {
+                instantValidator.validate(a.getIssueInstant(), "Country Assertion IssueInstant");
+                // FIXME: the expectedInResponseTo param is never used in the subjectValidator
+                subjectValidator.validate(a.getSubject(), a.getID());
+                conditionsValidator.validate(a.getConditions(), acceptableHubConnectorEntityIds.toArray(new String[0]));
+            } catch (Exception e) {
```

Comment:
Is it worth restricting this catch to the Exceptions we know that the validators throw, rather than catching everything?

Reply:
Agree, addressed with commit #2